antarest/study/business/binding_constraint_management.py

```diff
@@ -239,11 +250,26 @@ def get_binding_constraint(
             binding_constraint.append(new_config)
         return binding_constraint
 
+    def validate_binding_constraint(self, study: Study, constraint_id: str) -> None:
+        if int(study.version) < 870:
+            return  # There's nothing to check for constraints before v8.7
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        config = file_study.tree.get(["input", "bindingconstraints", "bindingconstraints"])
+        group = next((value["group"] for value in config.values() if value["id"] == constraint_id), None)
+        if not group:
+            raise BindingConstraintNotFoundError(study.id)
+        matrix_terms = {
+            "eq": get_matrix_data(file_study, constraint_id, "eq"),
```

Comment:
Actually, it's not necessary to check all the matrices: for the sake of efficiency, we could compare only the matrices corresponding to the type of binding constraint:

- if `operator == "equal"`, we check the "eq" matrix,
- if `operator == "greater"`, we check the "gt" matrix,
- if `operator == "less"`, we check the "lt" matrix,
- if `operator == "both"`, we check the "lt" and "gt" matrices.